xarray/core/dataset.py

```diff
@@ -4031,7 +4037,7 @@ def set_index(
                         f"dimension mismatch: try setting an index for dimension {dim!r} with "
                         f"variable {var_name!r} that has dimensions {var.dims}"
                     )
-                idx = PandasIndex.from_variables({dim: var})
+                idx = PandasIndex.from_variables({dim: var}, {})
```

Comment:
can we add the kwarg name for the second argument please? Or have `from_variables` accept `None`?